packages/bot-web-ui/src/components/flyout/flyout.jsx

```diff
@@ -47,7 +47,7 @@ const FlyoutContent = props => {
                 block: 'start',
             });
         }
-    }, [active_helper]);
+    }, [active_helper, flyout_ref]);
```

Comment:
Refs should not be added as a dependency because refs won't change only the value in `current` changes.

One more thing:
`const flyout_ref = React.createRef();`  
createRef is used in the function component which will create ref every time. It should be `useRef` instead.

Reply:
The reason we get the warning there is because ESLint is pretty limited in its ability to trace what you're doing with your JavaScript. So the React plugin for ESLint can't know that fly_out is actually a ref. So just to be safe, it warns us. Including refs in the dependency array doesn't make any difference anyway, I've just included it to avoid the warning that we're getting.

Reply:
Alright. But can we change `useRef` instead of `createRef`. 
We shouldn't be using `createRef` in a function component

Reply:
Changing the createRef to useRef fixed the warning and we do not need to add the ref to the dependency array( I removed the ref as dependency and the warning is fixed ). The following explanation of their differences seems useful for further usages.
The createRef is a function that creates a new ref every time. Unlike the useRef, it does not save its value between re-renders, instead creates a new instance of the ref for every re-render. Thus implying that it does not persist the existing ref between re-renders.

Reply:
I liked the communication here. Thank you @hirad-rewok for the explanation :+1: